unity/Holo/Assets/Scripts/ModelClippingPlaneControl.cs

```diff
@@ -0,0 +1,142 @@
+        HandTranslation.enabled = false;
+
+        ResetState();
+    }
+
+    public void Click(GameObject clickObj)
+    {
+        Debug.Log("Clicked obj: " + clickObj.name);
+        switch (clickObj.name)
+        {
+            case "ButtonClipping":
+                ClickedClipPlane();
+                break;
+
+            case "ButtonClippingTranslation":
+                ClickedClipPlaneTranslation();
+                break;
+
+            case "ButtonClippingRotation":
+                ClickedClipPlaneRotation();
+                break;
+        }
+
+    }
+
+    void ClickedClipPlaneTranslation()
+    {
+        Debug.Log("Toggle translation of clip plane");
+
+        ModelWithPlate.SetButtonState(ClippingPlaneRotationBtn, false);
+        if (ClippingPlaneState == ClipPlaneState.Translation)
+        {
+            clipPlaneQuadBbox.Deactivate();
+            HandTranslation.enabled = false;
+            ClippingPlaneState = ClipPlaneState.Active;
+            ModelWithPlate.SetButtonState(ClippingPlaneTranslationBtn, false);
+        }
+        else
+        {
+            clipPlaneQuadBbox.Activate();
+            HandTranslation.enabled = true;
+            ClippingPlaneState = ClipPlaneState.Translation;
+            ModelWithPlate.SetButtonState(ClippingPlaneTranslationBtn, true);
+        }
+    }
+
+    void ClickedClipPlaneRotation()
+    {
+        Debug.Log("Toggle rotation of clip plane");
+        HandTranslation.enabled = false;
+
+        ModelWithPlate.SetButtonState(ClippingPlaneTranslationBtn, false);
+        if (ClippingPlaneState == ClipPlaneState.Rotation)
+        {
+            clipPlaneQuadBbox.Deactivate();
+            ClippingPlaneState = ClipPlaneState.Active;
+            ModelWithPlate.SetButtonState(ClippingPlaneRotationBtn, false);
+        }
+        else 
+        {
+            clipPlaneQuadBbox.Activate();
+            ClippingPlaneState = ClipPlaneState.Rotation;
+            ModelWithPlate.SetButtonState(ClippingPlaneRotationBtn, true);
+        }
+    }
+
+    void ClickedClipPlane()
+    { 
+        if (!LoadedModel)
+        {
+            Debug.LogWarning("No model loaded for clipping plane");
+            ClippingPlaneState = ClipPlaneState.Disabled;
+            return;
+        }
+
+        SkinnedMeshRenderer modelRenderer = LoadedModel.GetComponent<SkinnedMeshRenderer>();
+
+        ClippingPlaneState = ClippingPlaneState == ClipPlaneState.Disabled ? ClipPlaneState.Active : ClipPlaneState.Disabled;
+
+        if (ClippingPlaneState == ClipPlaneState.Active)
+        {
+            modelRenderer.material = ClippingMaterial;
+            
+            ClippingPlaneTranslationBtn.gameObject.SetActive(true);
+            ClippingPlaneRotationBtn.gameObject.SetActive(true);
+            ModelWithPlate.SetButtonState(ClippingPlaneBtn, true); 
+        }
+        else
+        {
+            modelRenderer.material = DefaultModelMaterial;

```

Comment:
Instead of `DefaultModelMaterial`, I would call `GetComponent<ModelWithPlate>().MaterialNonPreview`. This avoids a possible bug that someone sets `ModelWithPlate.MaterialNonPreview` to one thing, but `ModelClippingPlaneControl.DefaultMaterial` to something else.

Reply:
done